update.sh

```diff
@@ -9,67 +9,63 @@ if pidof -o %PPID -x "update.sh" > /dev/null; then
+
+if [ -f "$compose_file" ]; then
+    echo "We have migrated to a docker compose setup to reduce risks caused by these update and run scripts. You can delete these scripts and use the Docker Compose workflow now."
+else 
+  wget -O "$compose_file" -T 10 -t 3 "https://raw.githubusercontent.com/filecoin-saturn/L1-node/main/docker-compose.yml"
+  if [[ -s "$compose_file" ]];
+  then
+    echo "Downloaded the docker compose file successfully!"
+  else
+    echo "Failed to download the docker compose file automagically. Please open a Github issue or migrate manually to the docker compose setup"
+    exit 1
+  fi
+
+  wget -O "$env_file" -T 10 -t 3 "https://raw.githubusercontent.com/filecoin-saturn/L1-node/main/.env"
+  if [[ -s "$env_file" ]];
+  then
+    echo "Downloaded the .env file successfully!"
+  else
+    echo "Failed to download the .env file automagically. Please open a Github issue or migrate manually to the docker compose setup"
+    exit 1
+  fi
+
+  echo "Testing compatibility with our Docker Compose workflow"
+  sudo docker compose version
+  if [ $? -ne 0 ];
+  then
+      echo "docker compose might not be supported by your setup, or you are still using an old docker-compose version";
+      exit 1
+  fi
+  echo "You have docker compose, proceeding."
+  echo "If any of the following steps fail, please edit the .env file to include the expected values and try running the commands:\n\t'sudo docker stop --time 900 saturn-node && sudo docker compose -f $compose_file up -d'."
+
+  random_sleep="$(( RANDOM % 2000 ))"
+  echo "Waiting for $random_sleep seconds to avoid stopping all nodes in the network at once..."
   sleep "$random_sleep"
 
-  echo -n "$(date -u) Draining $SATURN_NETWORK network L1 node... "
-  sudo docker kill --signal=SIGTERM saturn-node >> /dev/null
-  sleep 900
-  echo "restarting...."
+  echo "Pulling Saturn $SATURN_NETWORK network L1 node updates."
+  sudo -E docker compose -f $compose_file pull
 
-  sudo docker pull ghcr.io/filecoin-saturn/l1-node:$SATURN_NETWORK || true
-  sudo docker stop saturn-node || true
-  sudo docker rm -f saturn-node || true
-  sudo docker run --name saturn-node -it -d \
-    --restart=unless-stopped \
-    -v "$SATURN_HOME/shared:/usr/src/app/shared" \
-    -e "FIL_WALLET_ADDRESS=$FIL_WALLET_ADDRESS" \
-    -e "NODE_OPERATOR_EMAIL=$NODE_OPERATOR_EMAIL" \
-    -e "SPEEDTEST_SERVER_CONFIG=$SPEEDTEST_SERVER_CONFIG" \
-    --network host \
-    --ulimit nofile=1000000 \
-    ghcr.io/filecoin-saturn/l1-node:$SATURN_NETWORK
-  sudo docker image prune -f
+  echo "Draining $SATURN_NETWORK network L1 node... "
+  sudo docker stop --time 900 saturn-node || true
 
-  echo "Updated to latest version successfully!"
-else
-  echo "Saturn $SATURN_NETWORK network L1 node up to date"
+  echo "Restarting with docker compose now..."
+  sudo docker rm -f saturn-node || true
+  sudo -E docker compose -f $compose_file pull
```

Comment:
@AnomalRoil just to double-check: is the reason for this second pull to ensure we're running fresh images post-drain?

Reply:
Yes, because the `--time 900` in `docker stop` means it can take up to 15 minutes to shut down.